### PR TITLE
added function paddle.vision.transforms.crop and test draft

### LINKED ISSUE #21818: paddle.vision.transforms.crop

### PR DESCRIPTION
**Pull request by danieldaug**
Close #21818

### CODE REVIEW

File: ivy/functional/frontends/paddle/vision/transforms.py [hunk 1]
```diff
@@ -33,3 +33,11 @@ def vflip(img, data_format="CHW"):
     elif data_format.lower() == "hwc":
         axis = -3
     return ivy.flip(img, axis=axis)
+
+@with_unsupported_device_and_dtypes(
+    {"2.5.1 and below": ("float16", "float32", "int16", "int8", "uint8")}, "paddle"
+)
+@to_ivy_arrays_and_back
+def crop(img, top, left, height, width):
+    array = ivy.array(img)
```
Comment (AnnaTz) [inline ivy/functional/frontends/paddle/vision/transforms.py:42]: This conversion should be handled in `@to_ivy_arrays_and_back`, so I think calling `ivy.array` here is redundant.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/paddle/vision/transforms.py [hunk 2]
```diff
@@ -135,3 +135,10 @@ def vflip(img, data_format="CHW"):
     elif data_format.lower() == "hwc":
         axis = -3
     return ivy.flip(img, axis=axis)
+
+@with_unsupported_device_and_dtypes(
+    {"2.5.1 and below": ("float16", "float32", "int16", "int8", "uint8")}, "paddle"
+)
+@to_ivy_arrays_and_back
+def crop(img, top, left, height, width):
+    return img[top:top+height, left:left+width]
```
Comment (AnnaTz) [inline ivy/functional/frontends/paddle/vision/transforms.py:144]: Thanks for the update! Now the only thing remaining is to resolve the lint errors showing up in the CI check at the bottom of the PR.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/paddle/vision/transforms.py (hunk 2 above)
Comment (danieldaug) [reply ivy/functional/frontends/paddle/vision/transforms.py:144]: Thank for your reviewing and help so far! I am struggling to understand how to resolve these errors. More specifically, I am lost on how to find sufficient documentation that fully explains the parameters within the decorator '@handle_frontend_test' in order to make sure that the code there is correct as well as understanding how to navigate and debug the code based on the lint errors provided. Your help would be greatly appreciated!

Comment (AnnaTz) [reply ivy/functional/frontends/paddle/vision/transforms.py:144]: Resolving the lint errors is actually much simpler than what you are suggesting! 
Here what we can see in the CI check below:
image
The red and green highlights indicate the before and after of the changes you need to make. So, just add spaces in the index of `return img[top:top+height, left:left+width]` and add a `,` after `max_dim_size=100`. 
Lint errors are basically formatting errors in the code you are submitting.

File: ivy/functional/frontends/paddle/vision/transforms.py [hunk 3]
```diff
@@ -135,3 +135,10 @@ def vflip(img, data_format="CHW"):
     elif data_format.lower() == "hwc":
         axis = -3
     return ivy.flip(img, axis=axis)
+
```
Comment (AnnaTz) [inline ivy/functional/frontends/paddle/vision/transforms.py:138]: Please revert the changes in the `.idea` folder. These shouldn't get merged with your PR.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Comment by AnnaTz** [comment]
Hi @danieldaug, please resolve the conflicts so that the CI checks can properly run.

File: ivy/functional/frontends/paddle/vision/transforms.py [hunk 4]
```diff
@@ -137,10 +137,19 @@ def vflip(img, data_format="CHW"):
...
+
 @with_supported_dtypes(
     {"2.5.1 and below": ("float32", "float64", "int32", "int64")}, "paddle"
 )
 @to_ivy_arrays_and_back
 def hflip(img):
     img = ivy.array(img)
     return ivy.flip(img, axis=-1)
+
```
Comment (AnnaTz) [inline ivy/functional/frontends/paddle/vision/transforms.py:155]: The test seems to be passing, but there is a lint error in the CI checks. Please look into that.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_transforms.py [hunk 5]
```diff
@@ -3,7 +3,7 @@
 # local
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_transforms.py:3]: Also, the changes to the `.idea` folder have not been reverted yet.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_transforms.py (hunk 5 above)
Comment (danieldaug) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_transforms.py:3]: Hi again, I believe I have fixed the rest of the lint errors from earlier that you had mentioned in the last review. For the .idea folder reversion, I believe that the folder is the same as it was before my accidental change. If you check the difference in the files on Github, it shows one line being erased and added back in the exact same way it was before (not sure why). Will this be alright or is some other sort of action required so that this doesn't show up? Thank you for all your help over these past few days.

Comment (AnnaTz) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_transforms.py:3]: Hi @danieldaug, if you can't resolve this git-wise, you could just make a new PR where you copy-paste your intended changes only. Go ahead and tag me there to get it merged right away. Thanks!

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_transforms.py [hunk 6]
```diff
@@ -87,6 +87,7 @@ def test_paddle_vflip(
     test_flags,
     backend_fw,
 ):
+
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_transforms.py:90]: Hi @danieldaug, could you have a look at the lint tests again? There seem to be some errors.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_transforms.py (hunk 6 above)
Comment (danieldaug) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_transforms.py:90]: Hi @AnnaTz, I managed to resolve the lint flake8 error. Is it also necessary to resolve the black error? I'm not sure what that is addressing.

Comment (AnnaTz) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_transforms.py:90]: Hi @danieldaug, I think the black error comes up because you haven't resolved all errors.
image
There are some blank lines that need to be removed (red dash), and some that need to be added (green dash).

File: ivy/data_classes/container/experimental/utility.py [hunk 7]
```diff
@@ -1,5 +1,82 @@
...
+            Input container
+        out
+            Optional output container, for writing the result to.
+
+        Returns
+        -------
+        ret
+            Output container.
+        """
+        return self.static_optional_get_element(self, out=out)
```
Comment (AnnaTz) [inline ivy/data_classes/container/experimental/utility.py:82]: Not sure if this was here before, but what are these `optional_get` changes? It's not related to `crop` is it?

Comment (AnnaTz) [reply ivy/data_classes/container/experimental/utility.py:82]: Your main branch is not synced with ours.
image

Comment (AnnaTz) [reply ivy/data_classes/container/experimental/utility.py:82]: and you also seem to have pushed changes to the `test_array_api` files. We only merge PRs that make changes related to the task they are resolving. Anything extra needs to be removed.

Comment (danieldaug) [reply ivy/data_classes/container/experimental/utility.py:82]: I have synced my main branch although I am unable to access and reset or revert test_array_api as it is a subfolder containing 31 files that I have not touched. Could this problem be due to other users participating in my pull-request? Sorry for all the trouble, but all of these problems besides the lint errors have not been due to changes made by me so I am very confused as to why they are happening.

Comment (AnnaTz) [reply ivy/data_classes/container/experimental/utility.py:82]: I would suggest making a new branch for each task you start working on. This will make it easy to resolve such issues in the future. 
I think the most straightforward way to deal with this now is to 1) make a new fork, 2) create a branch and copy-paste your function and test there, 3) make a new PR from this branch.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.
